src/coreclr/gc/gc.cpp

```diff
@@ -12297,6 +12299,48 @@ void gc_heap::distribute_free_regions()
 {
 #ifdef USE_REGIONS
     const int kind_count = large_free_region + 1;
+    if (settings.reason == reason_induced_aggressive)
+    {
+#ifdef MULTIPLE_HEAPS
+        for (int i = 0; i < n_heaps; i++)
+        {
+            gc_heap* hp = g_heaps[i];
+            int hn = i;
+#else //MULTIPLE_HEAPS
+        {
+            gc_heap* hp = pGenGCHeap;
+            int hn  = 0;
+#endif //MULTIPLE_HEAPS
+            for (int kind = basic_free_region; kind < count_free_region_kinds; kind++)
+            {
+                global_regions_to_decommit[kind].transfer_regions (&hp->free_regions[kind]);
+            }
+            while (hp->decommit_step())
+            {
+            }
+            for (int i = 0; i < total_generation_count; i++)
+            {
+                generation* generation = hp->generation_of(i);
+                heap_segment* region = generation_start_segment (generation);
+                region = heap_segment_rw (region);
+                while (region != nullptr)
```

Comment:
nit - it's not conventional to write code like this. could you please change it to 

`heap_segment* region = heap_segment_rw (generation_start_segment (generation);`

?